### PR TITLE
fix appending 'dataloader_idx_i' breaks checkpoint name

### PR DESCRIPTION
**Pull request by richarddwang**
## What does this PR do?

When there are multiple validation dataloaders, `log` will append `/dataloader_idx_<i>` to the metric name. e.g. `self.log('accuracy')` results in metric name `accuracy/dataloader_idx_0`.

On the other hand, we can specify `filename` in `ModelCheckpoint` as `"name_{metric_name:.2f}"` and the `"{}"` will be replaced by the metric value according to the metric name.

Now the problem is when metric name is `"accuracy/dataloader_idx_0"`, and to get the value of it we specify `filename` of `ModelCheckpoint` as `"name_{accuracy/dataloader_idx_0:.2f}"`, the checkpoint will be saved as `"dataloader_idx_0=0.89.ckpt"` under a directory `"name_accuracy"`

So I change `"/dataloader_idx_<i>"` to `"(dataloader_idx_<i>)"` to solve this problem.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary) -> no need
- [x] Did you write any new necessary tests? (not for typos and docs) -> no need
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings) -> no need

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — CHANGES_REQUESTED** [review]
Hi @richarddwang 

I'm afraid we can't merge this, as it would be a breaking change for existing users who rely on this notation.

> Now the problem is when metric name is "accuracy/dataloader_idx_0", and to get the value of it we specify filename of ModelCheckpoint as "name_{accuracy/dataloader_idx_0:.2f}", the checkpoint will be saved as "dataloader_idx_0=0.89.ckpt" under a directory "name_accuracy"

You can avoid this problem by setting

```python
ModelCheckpoint(
    filename="name_(dataloader_idx_0){accuracy/dataloader_idx_0:.2f}"
    auto_insert_metric_name=False
)
```

**Comment by richarddwang** [comment]
Thanks. It is acceptable to me.
